.gitlab/kernel_version_testing/system_probe.yml

```diff
@@ -0,0 +1,76 @@
+.package_testsuites:
+  stage: kernel_version_testing
+  rules:
+    !reference [.on_system_probe_changes_or_manual]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/btf-gen:$DATADOG_AGENT_BTF_GEN_BUILDIMAGES
+  tags: ["runner:main"]
+  script:
+    - pushd $DD_AGENT_TESTING_DIR
+    - tar czvf $ARCHIVE_NAME site-cookbooks
+    - ls -lh .
+    - popd
+  artifacts:
+    when: always
+    paths:
+      - $DD_AGENT_TESTING_DIR/$ARCHIVE_NAME
+
+package_testsuites_x64:
+  extends:
+    - .package_testsuites
+  needs: ["tests_ebpf_x64"]
+  variables:
+    ARCHIVE_NAME: "site-cookbooks-x86_64.tar.gz"
+
+package_testsuites_arm64:
+  extends:
+    - .package_testsuites
+  needs: ["tests_ebpf_arm64"]
+  variables:
+    ARCHIVE_NAME: "site-cookbooks-arm64.tar.gz"
+
+
+kernel_version_testing_system_probe_linux_x64_ec2:
+  extends:
+    - .kitchen_ec2_location_us_east_1
+  stage: kernel_version_testing
+  rules:
+    !reference [.manual]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  needs: ["go_deps", "go_tools_deps", "package_testsuites_x64"]
+  tags: ["runner:main"]
+  variables:
+    ARCH: x86_64
+    EC2_INSTANCE_TYPE: "z1d.metal"
+    AWS_REGION: us-east-1
+    LibvirtSSHKeyX86: $CI_PROJECT_DIR/libvirt_rsa-x86
+    LibvirtSSHKeyARM: $CI_PROJECT_DIR/libvirt_rsa-arm
+  script:
+    - pulumi login s3://dd-ci-datadog-agent-omnibus-cache-build-stable/kernel-version-testing/$CI_PIPELINE_ID/pulumi
+    - inv -e system-probe.test-microvms --security-groups=$KITCHEN_EC2_SG_IDS --subnets=$KITCHEN_EC2_SUBNET --instance-type-x86=$EC2_INSTANCE_TYPE
+  artifacts:
+    when: always
+    paths:
+      - $LibvirtSSHKeyX86
+      - $LibvirtSSHKeyX86.pub
+      - $LibvirtSSHKeyARM
+      - $LibvirtSSHKeyARM.pub
+
+kernel_version_testing_system_probe_linux_x64_ec2_cleanup:
+  extends:
+    - .kitchen_ec2_location_us_east_1
+  stage: kernel_version_testing
+  rules:
+    !reference [.on_system_probe_changes_or_manual]
+  image: 486234852809.dkr.ecr.us-east-1.amazonaws.com/ci/datadog-agent-buildimages/system-probe_x64:$DATADOG_AGENT_SYSPROBE_BUILDIMAGES
+  needs: ["kernel_version_testing_system_probe_linux_x64_ec2"]
```

Comment:
I don't remember exact how this works, but does this job run even if the needed one fails?

Reply:
If I am not mistaken it will be skipped.

Reply:
We want it to run regardless of pass/fail right?

Reply:
I was thinking it would be better never to run this if the deployment of the micro-vms job fails. In-case the deployment fails midway, our monitors would eventually alert us of the stale instance, and we can investigate why the deployment failed initially.
However, if we always run the destroy job, we might not realize that there was some problem to begin with.

Reply:
Ah, I was thinking a test failure (meaning a functional test didn't pass) would cause the earlier job to fail, but I'm guessing that is not the case?